### DISCUSSION TITLE
Como começar na área BackEnd

### DISCUSSION THREAD

**Discussion by reginaldojunior** [backend-br/forum#125] [Geral]
Sugestões para quem começar na área backend ou desenvolvimento em geral.

**Comment by yanmagale** [comment]
@reginaldojunior boa pergunta. Vou tentar separar alguns tópicos para responder, acredito que não existe uma única resposta para esta pergunta.

## Aprendizado e Conhecimento Técnico 

Quando você inicia sua carreira na vida acadêmica, acredito que o caminho mais natural, que é o apresentado por 99% das faculdades e instituições hoje, é você seguir para o desenvolvimento backend.  Linguagens como Java, C#, C++, PHP e etc, são a grande quantidade de conteúdo que temos a disposição nas instituições acadêmicas. Então é um bom primeiro passo para se iniciar.

Caso você não tenha a possibilidade de iniciar por este caminho acadêmico, a internet esta rodeada de conteúdos de extrema qualidade e gratuitos, voltados para o ensino destas tecnologias. Vou indicar o canal do Guanabara, o [Curso em Vídeo](https://www.youtube.com/user/cursosemvideo) , que tem um ótimo conteúdo em várias(e outras) linguagens que mencionei.

## Contato com a Comunidade 

É muito muito muito importante que você tenha contato com os profissionais e o ecossistema daquela tecnologia/linguagem/ferramenta que você queira aprender. Além de conhecer as pessoas e suas histórias, você vai descobrir o que esta sendo utliizado no mercado, o que as pessoas estão falando sobre aquela tecnologia que você esta tentando aprender ... Então participe dos encontros que existirem na sua região, você pode achar vários grupos destes no [Meetup](https://www.meetup.com/pt-BR/). 

Nestes encontros, você também acaba conhecendo as empresas que utilizam aquela determinada tecnologia, como é o seu ecossistema, cultura e etc ... e frequentemente acabam surgindo oportunidades de emprego nestes encontros. 

Caso você tenha a oportunidade, recomendo participar de eventos fora da sua cidade, para que você também possa conhecer outros ecossistemas, locais e pessoas.

Falei um pouco sobre este assunto neste [vídeo](https://www.facebook.com/GoVulpi/videos/1940303019574329/)

## Projetos Pessoais e Open Source

Uma outra sugestão, é que você pratique e mostre o seu trabalho para as pessoas, das tecnologias que você esta estudando ou que deseja aprender mais, seja com projetos pessoais, ou ajudando os projetos que já existem aqui no Github



Espero ter ajudado 😄

**Comment by HermesMartins** [comment]
@yanmagale falou tudo. Eu, como sou mais visual, gosto bastante do Udemy para aprender uma tecnologia nova, lembrando que direto rola promoção então um curso de R$200~R$400 sai por menos de R$35. 

Outro ponto a se destacar são os projetos pessoais, é ai que você poem a cabeça para pensar e começa a aplicar o que você está aprendendo e não precisa ser nada complicado, por exemplo, um to-do list, um aplicativo de controle financeiro, etc... E isso faz uma grande diferença na hora de conseguir um emprego, você ter o que mostrar acaba sendo um grande diferencial.

Abs

**Comment by onhernandes** [comment]
[Esse artigo](https://medium.com/coderbyte/a-guide-to-becoming-a-full-stack-developer-in-2017-5c3c08a1600c) pode ajudar.

**Comment by nicholasess** [comment]
# Overview

Antes de começar, você precisa se familiarizar com algumas diretrizes que fazem a diferença no futuro.
Ler e entender a profissão front end, back end, DevOps por alto, não necessário botar a mão na massa. Identificar qual é o seu forte, visual ou algo mais underground, somente linhas de código. Nisso você traçará seus primeiros passos, front end para quem é mais visual e back end/DevOps para quem busca algo mais preto no branco. E se num futuro você vire Fullstack, que bom, mas antes não coloque o burro na frente da carroça.

Independente da linguagem que começar, aprender via lógica de programação te ajudará naqueles momentos que não importa a linguagem, seu raciocínio é seu melhor amigo. Muitos dizem pra não estudar, mas quando você aprende teste de mesa e outras coisas que só beneficiam, separa os homens dos garotos.

# Ler, ler, ler, ler...
Sua ansiedade te colocará numa posição para botar a mão na massa, mas ler muitos tutoriais sobre o mesmo assunto, te ajuda na compilação do entendimento como todo. Ler a documentação também é importante, pois, existe muitas coisas lá que não aparecem em tutoriais.

# Treinar muito
Um perigo eminente é achar que ler muito, fará você expert no assunto. Pode até ser em teoria, mas o que acontece na prática vai muito além de Getting Started. Treine em diversos cenários, se não souber, procure alguns para tal. Antes precavido do que surpreso por algo ter saído do controle.

# Sua linguagem favorita
Não adianta estar trabalhando numa linguagem que te dá dinheiro, todos os dias acordará com uma insatisfação, então encontre a linguagem que te faça bem. Pois, em dias atípicos, você estará estudando em como fazer o mesmo de forma diferente.

# Ser sociável
Não tanto assim, mas o possível para abrir portas e se diferenciar dos demais. Entre em grupos do Facebook ou encontre slacks que abordam os assunto que te interessam. Esteja ciente que ser sociável, te deixará numa zona que muitas oportunidades podem surgir, além de conhecer muitas pessoas.

# Colabore
Sabe quando alguém te pede uma ajuda? Independente do que for, ajude, mesmo que não saiba (utilizo essa tática). E não precisa colaborar sempre, quando você não souber mesmo e não faz ideia por onde começar, fique tranquilo e espere alguém fazer isso por você. O interessante da colaboração, é que você acaba aprendendo muito mais.

# Síndrome do Impostor
 "Não sou capaz, preciso estudar ainda mais". Porém, quando você está numa situação que precisa estudar, com certeza consegue fazer o que é necessário. Existem duas situações, aquela que você sabe que pode e aquela que você não faz a minima ideia e acha que pode. A que você pode, vai fundo, mas quando não sabe por onde começar, seja franco contigo e com quem está passando o desafio.

# Há vida lá fora
Mantenha sua vida fora do trabalho, dos estudos...Se alimente, faça atividade física, viaje, viva de verdade. Trabalho te ajuda a prover os prazeres, mas existem outros que não custa um real.

Abraços
